magma/circuit.py

```diff
@@ -58,6 +59,18 @@ class _SyntaxStyle(enum.Enum):
 class DefinitionContext:
     def __init__(self, placer):
         self.placer = placer
+        self._inline_verilog = []
+
+    def add_inline_verilog(self, format_str, format_args):
+        self._inline_verilog.append((format_str, format_args))
+
+    def finalize(self, defn):
+        final_placer = self.placer.finalize(defn)
+        final_context = DefinitionContext(final_placer)
+        with _DefinitionContextManager(final_context):
+            for format_str, format_args in self._inline_verilog:
+                defn.inline_verilog(format_str, **format_args)
```

Comment:
if we're deprecating `cls.inline_verilog()` then this codepath should ideally not use it. I would ideally move all the logic in `_inline_verilog` to an external function in `inline_verilog.py` that does the same thing but then all the logic is one file. And the deprecated `inline_verilog()`calls to that same function. Avoids the deprecation warning on the new API.

Reply:
Most of this logic is moved out of the file in the inline-env PR so I'll finish the cleanup there.